beluga/include/beluga/algorithm/distance_map.hpp

```diff
@@ -92,6 +92,31 @@ auto nearest_obstacle_distance_map(
   return distance_map;
 }
 
+/// A distance map function considering unknown values.
+template <class OccupancyGrid, class DistanceFunction, class NeighborsFunction>
+auto nearest_distance_map(
+    const OccupancyGrid& grid,
+    DistanceFunction&& distance_function,
+    NeighborsFunction&& neighbors_function,
+    const double& unknown_distance_map,
+    const bool& explore_unknown) {
+  auto obstacle_map = grid.obstacle_data();
+
+  auto distance_map = nearest_obstacle_distance_map(obstacle_map, distance_function, neighbors_function);
+
+  if (explore_unknown) {
+    auto uknown_map = grid.unknown_data();
+
+    for (auto [index, is_unknown] : ranges::views::enumerate(uknown_map)) {
+      if (is_unknown) {
+        distance_map[index] = unknown_distance_map;
+      }
+    }
+  }
+
+  return distance_map;
+}
```

Comment:
@DPR00 meta: I'm on the fence about having a separate function with a similar name but not quite, taking a grid instead of a range. Following the idea of the mask below, I feel that something like:

```c++
  auto distance_map = nearest_obstacle_distance_map(grid.obstacle_mask(), distance_function, neighbors_function);
  if (params_.model_unknown_space) {
      const double background_distance = /* compute it */ ;
      distance_map |= beluga::actions::overlay(grid.unknown_mask(), background_distance);
  }
```
would much more generally applicable. I can see the notion of a mask or stencil coming up again later on. How do you feel about implementing a new action? [`beluga::actions::normalize`](https://github.com/Ekumen-OS/beluga/blob/main/beluga/include/beluga/actions/normalize.hpp) is a good reference for this.

Reply:
I think I got the idea. It looks cleaner and much more generally applicable, so I will implement the action.

Reply:
PR updated :)